src/foam/u2/layout/Layouts.js

```diff
@@ -1,23 +1,207 @@
+  ],
+
+  css: `
+    ^ {
+      display: flex;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'Map',
+      name: 'defaultConfig'
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'border',
+      value: { class: 'foam.u2.borders.NullBorder' }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.u2.layout.AlignmentTypes',
+      name: 'horizontalAlignmentTypes',
+      documentation: `
+        An enum on horizontal alignment types
+        (i.e. on the web, this follows the 'justify-content' CSS property)
+      `,
+      value: null,
+    },
+    {
+      class: 'String',
+      name: 'horizontalAlignmentValue',
+      documentation: `
+        To render the proper CSS for web based on the alignment types enum.
+        Will later be used to also account for mobile.
+
+        NOTE: Added support for now to hedge for the future, but we will
+        use the direct flex columning as shown in foam.comics.v2.Comics for now
+      `,
+      expression: function(horizontalAlignmentTypes$webFlexProp){
+        return horizontalAlignmentTypes$webFlexProp;
+      }
+    }
+  ],
+
   methods: [
     function add() {
-      console.log('Adding column!');
-      return this.SUPER.apply(this, arguments);
+
+      /**
+       * Here we are checking for two cases
+       * 1. If a Col element is added, then call the super on it
+       * 2. If another element is added, then wrap it in a Col element with default configuration
+       */
+      [...arguments].forEach(value => {
+        if ( this.Col.isInstance(value) ) {
+          this.SUPER(value);
+        }
+        else {
+          // conditionally add isAlignmentAdded to defaultConfig if horizontal alignment was specified
+          if (this.horizontalAlignmentTypes){
+            this.defaultConfig.isAlignmentActivated = true;
+          }
+
+          this
+            .start(this.Col, this.defaultConfig)
+              .start(this.border)
+                .add(value)
+              .end()
+            .end();
+        }
+      })
+      return this;
+    },
+
+    function initE() {
+      this.SUPER();
+
+      // if horizontal alignment was specified
+      if (this.horizontalAlignmentValue) {
+        this.addClass(this.myClass())
+          .style({ 'justify-content': this.horizontalAlignmentValue$ });
+      } else {
+        this.addClass(this.myClass())
+      }
     }
   ]
+});
+
+foam.CLASS({
+  package: 'foam.u2.layout',
+  name: 'Col',
+  extends: 'foam.u2.Element',
+
+  documentation: `
+    An individual column element within a group of columns.
+  `,
+
+  properties: [
+    {
+      class: 'Float',
+      name: 'flex',
+      documentation: `
+        Define how much this column will grow (take up space) relative to the other columns within the Cols element
+      `,
+      expression: function(isAlignmentActivated){
+        return isAlignmentActivated ? 0 : 1;
+      }
+    },
+    {
+      class: 'Boolean',
+      name: 'isAlignmentActivated',
```

Comment:
Let's ditch this property completely and just default flex to 0.

Reply:
Done. Setting the default to 1 because if all columns were 0, then it would be right text right next to eachother. But Setting it to 1 actually divides the rows into n columns.

Reply:
^ Rescinding the previous comment, as we will make justify-content the default rows and cols thing